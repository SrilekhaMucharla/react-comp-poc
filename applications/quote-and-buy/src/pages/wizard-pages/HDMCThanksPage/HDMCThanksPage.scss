// this file taken from develop branch HDMCThanksPage.scss
// most classes can be removed after merge between develop and tealium branch happens
// but some are for multi car only

.mc-thanks-page__container-single-car-info {
    width: 596px;
    margin: 0px;
    height: 55px;
    @include media-breakpoint-down(sm) {
        height: 74px;
        width: 301px;
    }
}

.mc-thanks-page__mc-ref-number {
    line-height: 1.2;
    color: $link-color;
    margin-left: $tiny-value;
}

.mc-thanks-page__container-mc-ref {
    font-size: $md-value;
    margin-top: 14px;
    margin-bottom: $md-value;
    @include media-breakpoint-down(sm) {
        flex-direction: column;
    }
}

.mc-thanks-page__car-info-hr-line {
    margin: 0 11px;
}

.mc-thanks-page__container-registration {
    display: flex;
    align-items: center;
    padding-left: 11px;
    flex-grow: 1.5;
    @include media-breakpoint-down(sm) {
        flex-direction: column;
        align-items: end;
        justify-content: center;
    }
}

.mc-thanks-page__centered-row {
    display: flex;
    align-items: center;
}

.mc-thanks-page__car-model-info {
    margin-left: $tiny-value;
    @include media-breakpoint-down(sm) {
        margin-left: 0;
        margin-top: $tiny-value;
    }
}

.mc-thanks-page__container-policy {
    padding-right: 11px;
    justify-content: flex-end;
    @include media-breakpoint-down(sm) {
        flex-direction: column;
        align-items: end;
        justify-content: center;
    }
}

// this file needs to be removed after latest HDThanksPage.scss is pulled from develop

.mc-thanks-page {
    margin-top: 80px;
    .btn-lg,
    .btn-group-lg > .btn {
        width: inherit;
        color: $app-theme !important;
        border: 1px solid $mint-color !important;
        background-color: $green-transparent-20 !important;
        @include media-breakpoint-between(xs, sm) {
            width: 94%;
        }
    }
    @include media-breakpoint-down(md) {
        margin-top: 70px;
        max-width: none;
        padding: 0;
    }
}

.mc-thanks-page__arc-header {
    height: 360px;
    @include media-breakpoint-down(sm) {
        height: 380px;
        left: 0%;
        width: 100%;
    }
    background-color: $app-theme;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    z-index: 0;

    &:after {
        content: " ";
        background-color: $gray-light-8;
        position: absolute;
        bottom: 0;
        left: 0;
        width: 100%;
        height: 10%;
        border-radius: 100%/30px 30px 0 0;
        // clip-path: ellipse(42% 28px at 55.6% 103%);
        z-index: -1;
        @include media-breakpoint-down(sm) {
            // clip-path: ellipse(56% 28px at 55.6% 103%);
        }
    }
}

.mc-thanks-page__container-header {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    color: $white-color;
    @include media-breakpoint-down(sm) {
        display: contents;
    }
}

.mc-thanks-page__header-icon {
    background-color: $link-color;
    text-align: center;
    width: 50px;
    height: 50px;
    border-radius: 50%;
    z-index: 1;
    margin:0 auto;
    .fa-check {
        top: 10px;
        right: 1px;
        position: relative;
        font-weight: 600;
    }
}

.mc-thanks-page__header {
    z-index: 1;
    @include media-breakpoint-down(sm) {
        margin-top: 26px;
        margin-bottom: 7px;
    }
}

.mc-thanks-page__sub-header {
    font-size: $font-size-md;
    line-height: 1.5;
    z-index: 1;
    margin-top: 0px;
}

.mc-thanks-page__container-car-info {
    display: flex;
    justify-content: center;
    align-items: flex-end;
    text-align: left;
    background-color: $white-fill;
    color: $app-theme;
    margin: 30px auto 0 auto;
    border-radius: 7px;
    box-shadow: 0 15px 20px 0 rgba(0, 0, 0, 0.05);
    z-index: 1;
    width: 626px;
    @include media-breakpoint-down(sm) {
        width: 335px;
        margin-top: 30px;
    }
}

.mc-thanks-page__car-registration {
    border-radius: 2px;
    background-color: $yellow-color;
    text-align: center;
    width: 35%;
    font-size: 20px;
    font-weight: bold;
    line-height: 1.35;
    color: $app-theme;
    @include media-breakpoint-down(sm) {
        width: 75%;
    }
}

.mc-thanks-page__car_modal-info {
    font-weight: bold;
    font-size: $font-size-md;
    line-height: 1.2;
    margin-top: 10px;
}

.mc-thanks-page__car-policy-text {
    font-size: 13px;
    line-height: 1.5;
    // @include media-breakpoint-down(sm) {
    //     font-size: 13px;
    // }
}

.mc-thanks-page__car-policy-number {
    font-weight: bold;
    font-size: 12px;
    line-height: 1.2;
    color: $link-color;
    margin-left: 10px;
    @include media-breakpoint-down(sm) {
        margin-left: 0;
        margin-top: 8px;
    }
}

.mc-thanks-page__body {
    display: flex;
    text-align: left;
    margin-top: 31px;
    margin-bottom: 76px;
    color: $app-theme;
    @include media-breakpoint-down(sm) {
        display: contents;
    }
}

.mc-thanks-page__body-left {
    height: fit-content;
    box-shadow: 0 15px 20px 0 rgba(0, 0, 0, 0.05);
    padding: 0 30px 40px 30px;
    border-radius: 6px;
    background-color: $white-fill;
    width: 55%;
    @include media-breakpoint-between(xs, sm) {
        width: 100%;
    }
}

.mc-thanks-page__you-drive-logo-container {
    margin-left: -30px;
    margin-right: -30px;
    margin-bottom: 30px;

    .col {
        padding: 0;
    }
    .mc-thanks-page__you-drive-logo {
        width: 100%;
    }
}

.mc-thanks-page__container-left-steps {
    display: flex;
    flex-direction: row;
}

.mc-thanks-page__step-number {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 21px;
    height: 21px;
    padding: 10px;
    margin-top: 7px;
    border-radius: 50%;
    background-color: $alert-color;
    color: $white-color;
    font-size: $font-size-md;
    line-height: 1.2;
    font-weight: bold;
    position: relative;
    bottom: 6px;
}

.mc-thanks-page__step-header-hint {
    z-index: 1;
    margin-left: 10px;
}

.mc-thanks-page__store-images {
    flex-flow: row;
    display: inline-flex;
    img {
        height: auto;
    }
    img:nth-child(2) {
        margin-left: 5px;
    }
}

.mc-thanks-page__youdrivetabkey-image {
    margin-top: 30px;
    img {
        height: auto;
    }
}

.mc-thanks-page__container-managepolicy {
    @include media-breakpoint-down(sm) {
        margin-bottom: 45px;
    }
}

.mc-thanks-page__managepolicy-header {
    font-size: 20px;
    line-height: 1.2;
    @include media-breakpoint-down(sm) {
        margin-bottom: 8px;
    }
}

.mc-thanks-page__managepolicy-steps {
    margin-top: 13px;
    display: flex;
    flex-direction: row;
}

.mc-thanks-page__step-check-icon {
    padding: 0;
    color: $white-color;
    background-color: $link-color;
    width: 22px;
    height: 24px;
    border-radius: 50%;
    text-align: center;
    .fa-check {
        position: relative;
        top: 2px;
    }
}

.mc-thanks-page__managepolicy-step {
    font-weight: normal;
    color: #02213e;
}

.mc-thanks-page__managepolicy-button {
    padding: 30px 0;
    @include media-breakpoint-down(sm) {
        padding: 35px 0 0 0;
    }
}

.mc-thanks-page__body-right {
    height: fit-content;
    box-shadow: 0 15px 20px 0 rgba(0, 0, 0, 0.05);
    border-radius: 6px;
    background-color: $white-fill;
    width: 40%;
    margin-left: 2%;

    .hd-steps-container {
        margin: 0;
    }

    .hd-steps-container .hd-quote {
        font-size: $hd-font-size-h2-mobile !important;
    }

    @include media-breakpoint-between(xs, sm) {
        width: auto;
        margin-left: 0;
        margin-bottom: 60px;
    }
    .hd-description-label {
        position: relative;
        // top: 8px;
    }
    &-payment-breakdown {
        .hd-steps-col-container {
            .hd-step-row {
                &:last-child {
                    margin-top: 90px;
                }
            }
        }
        .hd-steps__col-container__row__description-col {
            .hd-description-name {
                margin-top: 10px;
                & > p {
                    margin-bottom: 0;
                }
            }
        }
    }
}

.mc-thanks-page__log-in-button {
    height: 50px;
    padding: 18px 23px;
    @include media-breakpoint-up(md) {
        height: 60px;
    }
    &:hover {
        padding: 18px 23px;
    }
}

.mc-thanks-page__sync-icon {
    &::before {
        color: $blue-secondary;
        position: relative;
        font-size: 23px;
        top: 3px;
    }
}

.mc-thanks-page__mobile-hand-image {
    position: relative;
    z-index: 0;
    right: 0;
    @include media-breakpoint-down(md) {
        position: absolute;
    }
    img {
        height: auto;
        width: 75%;
    }
}

.mc-thanks-page__container-left-steps-column {
    flex-flow: row;
}

.mc-thanks-page__my-account {
    color: $app-theme !important;
}

.mc-thanks-page__overlay__link {
    padding: 0;
    margin: 0;
    color: $app-theme;
    line-height: 1.3;
}
