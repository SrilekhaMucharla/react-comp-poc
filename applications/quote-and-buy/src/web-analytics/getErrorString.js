import {
    UW_ERROR_CODE,
    GREY_LIST_ERROR_CODE,
    CUE_ERROR_CODE,
    QUOTE_DECLINE_ERROR_CODE,
    SESSION_TIMEOUT_ERROR,
    VALIDATION_ERROR,
    QUOTE_RATE_ERROR_CODE
} from '../constant/const';

const BACKEND = 'backend';
const SESSION_TIMEOUT = 'session timeout';
const VALIDATION = 'validation';

const getErrorString = (errorCode) => {
    switch (errorCode) {
        case UW_ERROR_CODE: return BACKEND;
        case GREY_LIST_ERROR_CODE: return BACKEND;
        case CUE_ERROR_CODE: return BACKEND;
        case QUOTE_DECLINE_ERROR_CODE: return BACKEND;
        case QUOTE_RATE_ERROR_CODE: return BACKEND;
        case SESSION_TIMEOUT_ERROR: return SESSION_TIMEOUT;
        case VALIDATION_ERROR: return VALIDATION;
        default: return null;
    }
};

export default getErrorString;
