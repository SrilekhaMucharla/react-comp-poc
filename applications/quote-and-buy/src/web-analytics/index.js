import {
    HDModal,
    HDButtonRefactor,
    HDButtonDashed,
    HDToggleButtonGroupRefactor,
    HDDropdownList,
    HDTextInputRefactor,
    HDOverlayPopup,
    HDAsyncSelect,
    HDDatePickerRefactor,
    HDSwitch,
    HDTable,
    HDCheckbox,
    HDCompletedCardInfo,
    HDRadioButtonList,
    HDCheckboxButtonList,
    HDQuoteDownloadRefactor,
    HDLabelRefactor,
    HDPolicySelect
} from 'hastings-components';
import withEventTracking, { ACTION, COMPONENT_TYPE } from './withEventTracking';
import withViewTracking from './withViewTracking';


export const AnalyticsHDButton = withEventTracking(HDButtonRefactor, COMPONENT_TYPE.BUTTON, [ACTION.CLICK]);

export const AnalyticsHDButtonDashed = withEventTracking(HDButtonDashed, COMPONENT_TYPE.BUTTON, [ACTION.CLICK]);

export const AnalyticsHDToggleButtonGroup = withEventTracking(HDToggleButtonGroupRefactor, COMPONENT_TYPE.BUTTON_GROUP, [ACTION.CHANGE]);

export const AnalyticsHDDropdownList = withEventTracking(HDDropdownList, COMPONENT_TYPE.DROPDOWN, [ACTION.CHANGE]);

export const AnalyticsHDAsyncSelect = withEventTracking(HDAsyncSelect, COMPONENT_TYPE.ASYNC_DROPDOWN, [ACTION.CHANGE]);

export const AnalyticsHDTextInput = withEventTracking(HDTextInputRefactor, COMPONENT_TYPE.INPUT, [ACTION.FOCUS]);

export const AnalyticsHDDatePicker = withEventTracking(HDDatePickerRefactor, COMPONENT_TYPE.DATE_PICKER, [ACTION.CHANGE]);

export const AnalyticsHDSwitch = withEventTracking(HDSwitch, COMPONENT_TYPE.SWITCH, [ACTION.CHANGE]);

export const AnalyticsHDTable = withEventTracking(HDTable, COMPONENT_TYPE.TABLE, [ACTION.SELECT]);

export const AnalyticsHDCheckbox = withEventTracking(HDCheckbox, COMPONENT_TYPE.CHECKBOX, [ACTION.CHANGE]);

export const AnalyticsHDCompletedCardInfo = withEventTracking(HDCompletedCardInfo, COMPONENT_TYPE.CARD, [ACTION.EDIT, ACTION.DELETE]);

export const AnalyticsHDRadioButtonList = withEventTracking(HDRadioButtonList, COMPONENT_TYPE.RADIO_BUTTON_LIST, [ACTION.CHANGE]);

export const AnalyticsHDCheckboxButtonList = withEventTracking(HDCheckboxButtonList, COMPONENT_TYPE.CHECKBOX_BUTTON_LIST, [ACTION.CHANGE]);

export const AnalyticsHDQuoteDownload = withEventTracking(HDQuoteDownloadRefactor, COMPONENT_TYPE.BUTTON, [ACTION.CLICK]);

export const AnalyticsHDLabel = withEventTracking(HDLabelRefactor, COMPONENT_TYPE.LABEL, [ACTION.CLICK]);

export const AnalyticsHDModal = withViewTracking(
    withEventTracking(HDModal, COMPONENT_TYPE.MODAL, [ACTION.CONFIRM, ACTION.CANCEL, ACTION.OPEN]),
    COMPONENT_TYPE.MODAL
);

export const AnalyticsHDOverlayPopup = withViewTracking(
    withEventTracking(HDOverlayPopup, COMPONENT_TYPE.OVERLAY, [ACTION.CONFIRM, ACTION.CANCEL, ACTION.OPEN]),
    COMPONENT_TYPE.OVERLAY
);

export const AnalyticsHDPolicySelect = withViewTracking(
    withEventTracking(
        HDPolicySelect,
        COMPONENT_TYPE.BUTTON_GROUP,
        [ACTION.SELECT]
    )
);
