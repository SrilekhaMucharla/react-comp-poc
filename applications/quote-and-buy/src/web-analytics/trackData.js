/* eslint-disable no-undef */
export const trackView = (dataObj) => {
    if (typeof utag !== 'undefined') {
        utag.view(dataObj);
    }
};

export const trackEvent = (dataObj) => {
    if (typeof utag !== 'undefined') {
        utag.link(dataObj);
    }
};
