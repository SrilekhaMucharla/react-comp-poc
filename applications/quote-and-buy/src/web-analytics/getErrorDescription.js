import getErrorString from './getErrorString';

const BACKEND = 'backend';
const SESSION_TIMEOUT = 'session timeout';
const VALIDATION = 'validation';

const getErrorDescription = (errorCode) => {
    switch (getErrorString(errorCode)) {
        case BACKEND: return 'Backend error, covers error codes: 705, 716, 802, 805, 805.';
        case SESSION_TIMEOUT: return 'Session timeout error code: 408.';
        case VALIDATION: return 'Validation error.';
        default: return null;
    }
};

export default getErrorDescription;
