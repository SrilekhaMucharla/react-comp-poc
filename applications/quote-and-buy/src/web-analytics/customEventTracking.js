/* eslint-disable camelcase */
import { trackEvent } from './trackData';

const customEventTracking = (data) => {
    const {
        element_id, event_action, event_type, event_value, sales_journey_type
    } = data;
    trackEvent({
        ...data,
        element_id: (element_id) || 'custom-element',
        event_action: (event_action) || 'custom-value',
        event_type: (event_type) || 'custom_type',
        event_value: (event_value) || 'custom-label',
        sales_journey_type: (sales_journey_type) || 'single_car'
    });
};

export default customEventTracking;
