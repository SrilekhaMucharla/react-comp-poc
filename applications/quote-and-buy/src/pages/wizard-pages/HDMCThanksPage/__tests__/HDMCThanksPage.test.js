import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { HDDropdownList } from 'hastings-components';
import ViewModelServiceFactory from 'gw-portals-viewmodel-js/ViewModelServiceFactory';
import HDMCThanksPage from '../HDMCThanksPage';
import defaultTranslator from '../../__helpers__/testHelper';
import productMetadata from '../../../../generated/metadata/product-metadata.json';
import * as mcSubmission from '../../../../routes/mockMCSubmissionQuoted.json';
import * as multiCustomizeSubmission from '../../../../routes/mockMultiCustomizeSubmissionVM.json';
import withTranslator from '../../__helpers__/test/withTranslator';
import createPortalRoot from '../../__helpers__/test/createPortalRoot';
import mockPaymentSchedule from '../mock/mockMCPSnormal.json';
import { trackView, trackEvent } from '../../../../web-analytics/trackData';

const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useDispatch: () => jest.fn()
}));

jest.mock('../../../../web-analytics/useMultiCarJourney.js', () => () => true);

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => ({
        state: ''
    })
}));

jest.mock('../../../../web-analytics/trackData', () => ({
    trackEvent: jest.fn(),
    trackView: jest.fn()
}));

jest.mock('hastings-capability-renewaloptions', () => ({
    HastingsRenewalOptionService: {
        setRenewalOption: jest.fn().mockResolvedValue({
            result: {
                resultFlag: true
            }
        })
    }
}));
jest.mock('../MCPaymentBreakdownData.js', () => ({
    getMCYearlyPaymentBreakDownData: jest.fn().mockReturnValue([
        {
            id: 1,
            circle: {
                type: 'violet',
                date: { day: 1, shortMonth: '2' }
            },
            description: {
                label: ['TEST123'],
                name: ['TEST123'],
                quote: { single: '£12' },
                tooltip: <div />
            }
        }, {
            id: 2,
            circle: {
                date: { day: 1, shortMonth: '2' }
            },
            description: {
                name: ['TEST123'],
            }
        }, {
            id: 3,
            circle: {
                type: 'dot'
            },
            description: {
                name: ['TEST123'],
                quote: { single: `£${12}` }
            }
        }, {
            id: 7,
            circle: {
                date: { day: 1, shortMonth: '2' }
            },
            description: {
                tooltip: <div />
            }
        }
    ]),
    getMCMonthlyPaymentBreakDownData: jest.fn().mockReturnValue([
        {
            id: 1,
            circle: {
                type: 'violet',
                date: { day: 1, shortMonth: '2' }
            },
            description: {
                label: ['TEST123'],
                name: ['TEST123'],
                quote: { single: '£12' },
                tooltip: <div />
            }
        }, {
            id: 2,
            circle: {
                date: { day: 1, shortMonth: '2' }
            },
            description: {
                name: ['TEST123'],
            }
        }, {
            id: 3,
            circle: {
                type: 'dot'
            },
            description: {
                name: ['TEST123'],
                quote: { single: `£${12}` }
            }
        }, {
            id: 7,
            circle: {
                date: { day: 1, shortMonth: '2' }
            },
            description: {
                tooltip: <div />
            }
        }
    ])
}));

const additionalEventValues = {
    customer_id: 'XA,XA,XA',
    insurance_product: 'HE,HD',
    product_option: 'Hastings Essential,Hastings Direct',
    product_option_code: 'HE,HD',
    sales_journey_type: 'multi_car',
    event_category: 'NEWBUS Car',
    journey_type: 'Car'
};

const expectedEventValues = [
    {
        order_reference_number: 'XA50000000279',
        policy_number: 'XA,XA,XA',
        sales_journey_type: 'multi_car'
    },
    {
        event_value: 'Change preference',
        event_type: 'label_click',
        element_id: 'gipp-nb-auto-rn-change-preference',
        event_action: 'Renewal preferences',
        ...additionalEventValues
    },
    {
        event_value: 'I think auto renewal is used to bump up the price of my quote',
        event_type: 'dropdown_change',
        element_id: 'gipp-nb-auto-rn-opt-out-dropdown',
        sales_journey_type: 'multi_car',
        event_action: 'Please tell us your reason for opting out...',
        ...additionalEventValues
    },
    {
        event_value: 'Confirm',
        event_action: 'Opting out',
        event_type: 'modal_confirm',
        element_id: 'gipp-nb-auto-rn-opt-out-confirm',
        optOutReasonCode: 'I think auto renewal is used to bump up the price of my quote',
        ...additionalEventValues
    },
    {
        event_type: 'toast_open',
        element_id: 'renewal-preference-toast',
        event_value: 'Toast',
        sales_journey_type: 'multi_car'
    }
];

const eventValues = [
    {
        event_value: 'Other',
        event_type: 'dropdown_change',
        element_id: 'gipp-nb-auto-rn-opt-out-dropdown',
        event_action: 'Please tell us your reason for opting out...',
        ...additionalEventValues
    },
    {
        event_value: 'Confirm',
        event_action: 'Opting out',
        event_type: 'modal_confirm',
        element_id: 'gipp-nb-auto-rn-opt-out-confirm',
        optOutReasonCode: 'Other',
        ...additionalEventValues
    },
    {
        event_type: 'toast_open',
        element_id: 'renewal-preference-toast',
        event_value: 'Toast',
        sales_journey_type: 'multi_car'
    }
];

const initializeStore = () => {
    const viewModelService = ViewModelServiceFactory.getViewModelService(
        productMetadata, defaultTranslator
    );

    const mcSubmissionVM = viewModelService.create(
        mcSubmission.result,
        'pc',
        'com.hastings.edgev10.capabilities.quote.submission.dto.HastingsMultiQuoteDataDTO'
    );

    const multiCustomSubVM = viewModelService.create(
        multiCustomizeSubmission.result,
        'pc',
        'com.hastings.edgev10.capabilities.quote.submission.dto.HastingsMultiCustomQuoteDTO'
    );

    const initialState = {
        wizardState: {
            app: {
                multiCarFlag: true
            },
            data: {
                mcsubmissionVM: mcSubmissionVM,
                multiCustomizeSubmissionVM: multiCustomSubVM
            }
        },
        mcPaymentScheduleModel: {
            mcPaymentScheduleObject: mockPaymentSchedule.result[0]
        }
    };

    return mockStore(initialState);
};

const pageMetadata = { page_name: 'test', page_type: 'test', sales_journey_type: 'multi_car' };

const mockProps = { pageMetadata: pageMetadata };

async function initializeWrapper(store, props) {
    let wrapper;
    createPortalRoot();
    await act(async () => {
        wrapper = mount(withTranslator(
            <Provider store={store}>
                <HDMCThanksPage {...props} />
            </Provider>
        ));
    });
    wrapper.update();
    return wrapper;
}

describe('<HDMCThanksPage />', () => {
    afterEach(() => {
        trackEvent.mockClear();
        trackView.mockClear();
    });
    test('render component', () => {
        const initialState = {};
        const store = mockStore(initialState);

        const wrapper = shallow(
            <Provider store={store}>
                <HDMCThanksPage pageMetadata={pageMetadata} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Component should have login button', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        expect(wrapper.find('.mc-thanks-page__log-in-button')).toBeTruthy();
    });

    test('Component should have container to show all quotes ref number', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        expect(wrapper.find('mc-thanks-page__container-car-info')).toBeTruthy();
    });
    test('should render overlay when optout link was clicked', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        const overlay = wrapper.find('HDRenewalPreference');
        const modal = wrapper.find('HDModal');
        expect(overlay.length).toBe(1);
        expect(modal.props().show).toBeTruthy();
    });
    test('should not render overlay when optout link was clicked and overlay confirmed', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        const dropdown = wrapper.find('HDDropdownList');
        const options = dropdown.prop('options');

        await act(async () => dropdown.props().onChange({ target: { value: options[0] } }));
        await act(async () => wrapper.update());
        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('Button').at(0).props().onClick({
            target: {
                value: {
                    label: 'Confirm'
                }
            }
        }));
        await act(async () => wrapper.update());
        expect(wrapper.find('HDModal').prop('show')).toBeFalsy();
    });
    test('should not render overlay when optout link was clicked and overlay canceled', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        await act(async () => wrapper.update());

        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('Button').at(1).props().onClick({
            target: {
                value: {
                    label: 'Cancel'
                }
            }
        }));
        wrapper.update();
        expect(wrapper.find('HDModal').prop('show')).toBeFalsy();
    });

    test('should send event when optout link was clicked', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        expect(trackEvent).toHaveBeenNthCalledWith(1, expectedEventValues[1]);
    });
    test('should send event when optout link was clicked and overlay confirmed', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        const dropdown = wrapper.find(HDDropdownList);
        const options = dropdown.prop('options');

        await act(async () => dropdown.props().onChange({ target: { value: options[0] } }));
        await act(async () => wrapper.update());
        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('Button').at(0).props().onClick({
            target: {
                value: {
                    label: 'Confirm'
                }
            }
        }));
        await act(async () => wrapper.update());
        expect(trackEvent).toHaveBeenNthCalledWith(2, expectedEventValues[2]);
        expect(trackEvent).toHaveBeenNthCalledWith(3, expectedEventValues[3]);
        expect(trackEvent).toHaveBeenNthCalledWith(4, expectedEventValues[4]);
    });
    test('should send opt out reason when optout other reason was selected and text fill', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        const dropdown = wrapper.find('HDDropdownList');
        const options = dropdown.prop('options');

        await act(async () => dropdown.props().onChange({ target: { value: options[3] } }));
        await act(async () => wrapper.update());

        const textArea = wrapper.find('textarea');
        await act(async () => textArea.props().onChange({
            target: {
                value: 'Test reason inside the areatext.'
            }
        }));
        await act(async () => wrapper.update());

        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('Button').at(0).props().onClick({
            target: {
                value: {
                    label: 'Confirm'
                }
            }
        }));
        await act(async () => wrapper.update());
        expect(trackEvent).toHaveBeenNthCalledWith(2, eventValues[0]);
        expect(trackEvent).toHaveBeenNthCalledWith(3, eventValues[1]);
        expect(trackEvent).toHaveBeenNthCalledWith(4, eventValues[2]);
    });
    test('should send opt out reason when optout other reason was selected and text not fill', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        const dropdown = wrapper.find('HDDropdownList');
        const options = dropdown.prop('options');

        await act(async () => dropdown.props().onChange({ target: { value: options[3] } }));
        await act(async () => wrapper.update());

        const textArea = wrapper.find('textarea');
        await act(async () => textArea.props().onChange({
            target: {
                value: ''
            }
        }));
        await act(async () => wrapper.update());

        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('Button').at(0).props().onClick({
            target: {
                value: {
                    label: 'Confirm'
                }
            }
        }));
        await act(async () => wrapper.update());
        expect(trackEvent).toHaveBeenNthCalledWith(2, eventValues[0]);
        expect(trackEvent).toHaveBeenNthCalledWith(3, eventValues[1]);
        expect(trackEvent).toHaveBeenNthCalledWith(4, eventValues[2]);
    });
    test('should send redacted opt out reason when optout other reason contains nums and/or email', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');
        const dropdown = wrapper.find('HDDropdownList');
        const options = dropdown.prop('options');

        await act(async () => dropdown.props().onChange({ target: { value: options[3] } }));
        await act(async () => wrapper.update());

        const textArea = wrapper.find('textarea');
        await act(async () => textArea.props().onChange({
            target: {
                value: 'test@mail1.com test123 123test te2st test'
            }
        }));
        await act(async () => wrapper.update());

        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('Button').at(0).props().onClick({
            target: {
                value: {
                    label: 'Confirm'
                }
            }
        }));
        await act(async () => wrapper.update());
        expect(trackEvent).toHaveBeenNthCalledWith(4, eventValues[2]);
    });

    test('should send wa event when optout modal close button is pressed', async () => {
        // setup
        const store = initializeStore();
        const wrapper = await initializeWrapper(store, mockProps);
        const tempStorage = expectedEventValues[1];
        expectedEventValues[1] = {
            element_id: 'gipp-nb-auto-rn-opt-out-close',
            event_action: 'Opting out',
            event_type: 'modal_close',
            event_value: 'close',
            ...additionalEventValues
        };

        // test
        const optoutlink = wrapper.find('.thanks-page__opt-out-link');
        optoutlink.at(0).simulate('click');

        await act(async () => wrapper.update());

        const modal = wrapper.find('HDModal');
        await act(async () => modal.find('[data-testid="close-button"]').at(0).props().onClick({
            target: {
                value: {
                    label: 'Confirm'
                }
            }
        }));
        await act(async () => wrapper.update());
        expect(trackEvent).toHaveBeenNthCalledWith(2, expectedEventValues[1]);

        // cleanup
        expectedEventValues[1] = tempStorage;
    });
});
