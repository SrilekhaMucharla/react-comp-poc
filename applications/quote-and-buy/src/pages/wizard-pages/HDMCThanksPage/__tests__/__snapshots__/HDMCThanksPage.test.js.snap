// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDMCThanksPage /> render component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(HDMCThanksPage)
    pageMetadata={
      Object {
        "page_name": "test",
        "page_type": "test",
        "sales_journey_type": "multi_car",
      }
    }
  />
</ContextProvider>
`;
